detox/test/e2e/17.datePicker.test.js

```diff
@@ -19,9 +19,20 @@ describe(':ios: DatePicker', () => {
       }
     });
 
-    it('can select dates on a UIDatePicker', async () => {
+    it(':ios: can select dates on a UIDatePicker', async () => {
       await element(by.id('datePicker')).setDatePickerDate('2019-02-06T05:10:00-08:00', "ISO8601");
       await expect(element(by.id('utcDateLabel'))).toHaveText('Date (UTC): Feb 6th, 2019');
       await expect(element(by.id('utcTimeLabel'))).toHaveText('Time (UTC): 1:10 PM');
     });
+
+    it(':android: can select dates on a UIDatePicker', async () => {
+      await element(by.id('openAndroidDatePicker')).tap();
+
+      //rn-datepicker does not support testId's on android, so by.type is the only way to match the datepicker right now
+      //@see https://github.com/react-native-datetimepicker/datetimepicker#view-props-optional-ios-only
+      await element(by.type('android.widget.DatePicker')).setDatePickerDate('2019-02-06T05:10:00-08:00', "ISO8601");
```

Comment:
<img width="1111" alt="image" src="https://user-images.githubusercontent.com/1962469/210088882-19454b05-1923-46a1-86e0-af7e00e1fdfc.png">

I see an error like that on CI:
https://buildkite.com/wix-mobile-oss/detox/builds/1871#018563a8-6f16-438e-ac69-4f1785fe8ec3/6-4104

Reply:
Weird. It worked on my machine. One possible explanation I can think of is, that detox's auto waiting might not work well together with how android updates the datepicker value.
This might cause the test to work on some systems, but not on others due to timing issues.

I will try to use a waitFor to verify this hypothesise.